packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4369,67 +4369,343 @@ class PositionedDirectional extends StatelessWidget {
+///   it's going to be. It means the space (or size) reserved for it
+///   is the maximum size the [Flexible] _could be,_ in the main axis.
+///
+///   Add up all the [flex] values of each [Flexible]'s fit parameter,
+///   and each [Flexible] will have a percentage of the leftover space
+///   reserved for it. This percentage will be equal to the percentage
+///   of total [flex] each [Flexible]'s [flex] value represents. IE:
+///
+///   [Flexible] 1 has [flex] of 50.
+///   [Expanded] 1 has [flex] of 30.
+///   [Expanded] 2 has [flex] of 20.
+///
+///   In the above case, the total of all flex parameters is 100. The
+///   [Flexible] will get 50/100 (50%) of the remaining space reserved
+///   for it. The first [Expanded] will get 30/100 (30%) reserved for it,
+///   and the second one will get 20% of the remaining space
+///   reserved for it.
+///
+///   The numbers you use for your [Flexible.fit] parameters
+///   don't have to add to 100. This example does so only because
+///   it makes understanding the process easier.
+///
+/// 3. Layout the [Flexible] and [Expanded] children, and use
+///     the same cross axis size limits (constraints) that were used
+///     in Step 1. Remember, if using [CrossAxisAlignment.stretch] then
+///     force all of the children to be max possible size in the cross axis,
+///     else let them be the size they want.
+///
+///     Unlike in Step 1, this time set _maximum_ main axis constraints
+///     (max size limits) for each [Flexible] and [Expanded] to be the
+///     sizes that were reserved for them in Step 2. In other words, the
+///     max possible main axis sizes were calculated in Step 2, and now
+///     those sizes get used when actually laying out each [Flexible]
+///     and [Expanded] in Step 3.
+///
+///     This next part is where the fit comes in.
+///
+///    Children with a fit property that is set to [FlexFit.tight] are given
+///    "tight constraints". As mentioned previously, this means the min
+///    size limit is set to be the same as the max size limit, and this
+///    gives the child only one size to be, since min == max.
+///
+///     _This will force the child of the [Expanded] to be the size specified
+///     by the tight constraints, no matter what size the child of the flexible
+///     might have been set to._ Any size you specify for the main axis
+///     of a widget that you then put in a [Flexible] with it's fit set to
+///     [FlexFit.tight] will be ignored, no matter if that size is larger or
+///     smaller than the leftover space. The child will always be as large
+///     as the space that was reserved for its parent [Expanded].
+///
+///    The only  children who should have a tight fit are the [Expanded]s.
+///    It is considered bad to use a [Flexible] and set its fit to tight
+///    because doing so creates unnecessary confusion, and this could be
+///    a reason for having one's coffee privileges revoked. If you need a
```

Comment:
This is a little confusing: why is it bad?

Also, the joke is funny, but let's leave it out, since jokes don't often translate well to other languages and cultures, and can confuse non-native speakers.